### PR TITLE
Move Tvserver to examples

### PR DESCRIPTION
**Pull request by xylophone21**
#### Problem
* Old CHIPTVServer is under src/android ( APK) and src/app/server ( zap ), this cause that we can not have a factory defined CHIPTVServer ( we may change the APK and zap defines )

#### Change overview
* Move app-server JNI shared lib and tv-common(zzz_generated/tv-app/zap-generated) link to example
* Move CHIPTVServer to example and let it build under example as other platforms

#### Testing
* Basic test CHIPTVServer with chip-tool

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking, given this is a platform maintainer updating their platform.

**Comment by xylophone21** [comment]
> Fast tracking, given this is a platform maintainer updating their platform.

@woody-apple so what should I do next on this PR?  Rebase to fix the "Tests / Test Suites" canceled issue or ask more people to review it? Thanks

**Review by lazarkov — APPROVED** [review]

File: examples/tv-app/android/java/TVApp-JNI.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+/**
+ *    @file
+ *      Implementation of JNI bridge for Tv App on Android.
```
Comment (andy31415) [inline examples/tv-app/android/java/TVApp-JNI.cpp:21]: nit: this `@file` comment does not seem to help me understand code more: it is obvious that a file named  `TVApp-JNI.cpp` contains an implementation of JNI for a TV APP

Comment (xylophone21) [reply examples/tv-app/android/java/TVApp-JNI.cpp:21]: OK, I will remove it in next PR

**Pull request merged**